### PR TITLE
Fix Darwin

### PR DESCRIPTION
**Pull request by ghost**
- Build CURL from source
- Set the zenity path on macOS
- Add zenity and OpenSSL to the list of dependencies

(Haven't tested the browser or UI stuff yet, this just fixes it so that you can launch and run the game on  macOS 13.3-- also not tested on Linux yet)

### CODE REVIEW

File: src/minecraft/std/string_darwin.cpp [hunk 1]
```diff
@@ -54,6 +54,6 @@ const char *mcpe::string::c_str() const {
 }
 
 std::ostream& operator<< (std::ostream& os, const mcpe::string& obj) {
-    os << obj.std();
+    os << obj.c_str();
```
Comment (mhsjlw) [inline src/minecraft/std/string_darwin.cpp]: Is this OK?

File: src/ui/file_picker/file_picker_zenity.cpp [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
 #include <unistd.h>
 #include <sys/wait.h>
 #include <sys/stat.h>
 #include <stdexcept>
 #include <cstring>
 #include "../../common/log.h"
 
+#ifdef __APPLE__
+// `brew install zenity`
+const std::string ZenityFilePicker::EXECUTABLE_PATH = "/usr/local/bin/zenity";
```
Comment (mhsjlw) [inline src/ui/file_picker/file_picker_zenity.cpp]: Can we grab this from the PATH instead, possibly?

File: README.md [hunk 3]
```diff
@@ -40,7 +42,7 @@ The commands to build using CMake are as follows:
 
 ```console
 $ mkdir build/ && cd build/
-$ cmake ..
+$ cmake -DOPENSSL_ROOT_DIR=/usr/local/opt/openssl ..
```
Comment (mhsjlw) [inline README.md]: This is now _required_

**Comment by ghost** [comment]
This isn't going to be updated or merged as we're moving to `minecraft-linux`. See macOS integration here: https://github.com/minecraft-linux/mcpelauncher-manifest/pull/1

**Pull request closed without merging**